django/db/migrations/loader.py

```diff
@@ -316,6 +319,34 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
 
+    def check_consistent_history(self, connection):
+        """
+        Checks that all migrations are applied after their dependencies. Raises
+        InconsistentMigrationHistory exception when migration in applied before
+        it's dependency
+        """
+        recorder = MigrationRecorder(connection)
+        applied = recorder.applied_migrations()
+        for migration in applied:
+            # If we don't know anything about this migration we skip it
+            if migration not in self.graph.nodes:
+                continue
+            # Check only parents from the same app
+            unapplied_parents = [
+                x.key for x in self.graph.node_map[migration].parents
+                if x not in applied and x[0] == migration[0]
+            ]
```

Comment:
That was just a quick fix for the issue, and I forgot about it. Without this check I keep getting `Migration admin.0001_initial is applied before its dependency contenttypes.0001_initial` or `Migration admin.0001_initial is applied before its dependency auth.0001_initial`, which is strange. I think the issue may be [here](https://github.com/django/django/blob/master/tests/runtests.py#L150-L156) and the possible solution can be to add `admin: None` to `settings.MIGRATION_MODULES` either in `runtests.py` or in `@override_settings`. I think editing `runtests.py` would be more natural, but I'm not sure if there could be some side effects to this.